api/types/authentication.go

```diff
@@ -701,3 +714,122 @@ func (d *MFADevice) MarshalJSON() ([]byte, error) {
 func (d *MFADevice) UnmarshalJSON(buf []byte) error {
 	return jsonpb.Unmarshal(bytes.NewReader(buf), d)
 }
+
+// type RequireMFATypeString string
+
+// const (
+// 	// RequireMFAOff means MFA is *not* required to begin server sessions.
+// 	RequireMFAOff RequireMFATypeString = "off"
+// 	// RequireMFASession means MFA is required to begin server sessions.
+// 	RequireMFASession RequireMFATypeString = "session_mfa"
+// 	// RequireMFASessionAndHardwareKey means MFA is required to begin server sessions,
+// 	// and login sessions must use a private key backed by a hardware key.
+// 	RequireMFASessionAndHardwareKey RequireMFATypeString = "hardware_key"
+// 	// RequireMFAHardwareKeyTouch means login sessions must use a hardware private key that
+// 	// requires touch to be used. This touch requirement applies to all API requests
+// 	// rather than only session requests. This touch is different from MFA, so to prevent
+// 	// requiring double touch on session requests, normal Session MFA is disabled.
+// 	RequireMFAHardwareKeyTouch RequireMFATypeString = "hardware_key_touch"
+// )
+
+// IsSessionMFARequired returns whether this RequireMFAType requires per-session MFA.
+func (r RequireMFAType) IsSessionMFARequired() bool {
+	return r == RequireMFAType_SESSION || r == RequireMFAType_SESSION_AND_HARDWARE_KEY
+}
+
+// MarshalJSON marshals RequireMFAType to boolean or string.
+func (r *RequireMFAType) MarshalYAML() (interface{}, error) {
+	val, err := r.encode()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return val, nil
+}
+
+// UnmarshalYAML supports parsing RequireMFAType from boolean or alias.
+func (r *RequireMFAType) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var val interface{}
+	err := unmarshal(&val)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = r.decode(val)
+	return trace.Wrap(err)
+}
+
+// MarshalJSON marshals RequireMFAType to boolean or string.
+func (r *RequireMFAType) MarshalJSON() ([]byte, error) {
+	val, err := r.encode()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return json.Marshal(val)
```

Comment:
```suggestion
	out, err := json.Marshal(val)
	return out, trace.Wrap(err)
```
So we trace wrap the marshal error